apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/FdaDrugCodeDisplayLookup.java

```diff
@@ -0,0 +1,179 @@
+   * have to construct it over and over in the unit tests
+   */
+  private static FdaDrugCodeDisplayLookup drugCodeLookupForTesting;
+
+  /**
+   * Cached copy of the production version of the {@link FdaDrugCodeDisplayLookup} so that we don't
+   * have to construct it over and over in the unit tests
+   */
+  private static FdaDrugCodeDisplayLookup drugCodeLookupForProduction;
+
+  /**
+   * Factory method for creating a {@link FdaDrugCodeDisplayLookup} for testing that includes the
+   * fake drug code.
+   *
+   * @return the {@link FdaDrugCodeDisplayLookup}
+   */
+  public static FdaDrugCodeDisplayLookup createDrugCodeLookupForTesting() {
+    if (drugCodeLookupForTesting == null) {
+      drugCodeLookupForTesting = new FdaDrugCodeDisplayLookup(true);
+    }
+
+    return drugCodeLookupForTesting;
+  }
+
+  /**
+   * Factory method for creating a {@link FdaDrugCodeDisplayLookup} for production that does not
+   * include the fake drug code.
+   *
+   * @return the {@link FdaDrugCodeDisplayLookup}
+   */
+  public static FdaDrugCodeDisplayLookup createDrugCodeLookupForProduction() {
+    if (drugCodeLookupForProduction == null) {
+      drugCodeLookupForProduction = new FdaDrugCodeDisplayLookup(false);
+    }
+
+    return drugCodeLookupForProduction;
+  }
+
+  /**
+   * Constructs an {@link FdaDrugCodeDisplayLookup}
+   *
+   * @param includeFakeDrugCode whether to include the fake testing drug code or not
+   */
+  private FdaDrugCodeDisplayLookup(boolean includeFakeDrugCode) {
+    readFDADrugCodeFile(includeFakeDrugCode);
+  }
+
+  /**
+   * Retrieves the Drug Code Display (SUBSTANCENAME) for the given Drug Code (PRODUCTNDC) using the
+   * drug code file downloaded during the build.
+   *
+   * @param claimDrugCode - NDC value in claim records
+   * @return the fda drug code display string
+   */
+  public String retrieveFDADrugCodeDisplay(Optional<String> claimDrugCode) {
+    /*
+     * Handle bad data (e.g. our random test data) if drug code is empty or length is less than 9
+     * characters
+     */
+    if (!claimDrugCode.isPresent() || claimDrugCode.isEmpty() || claimDrugCode.get().length() < 9) {
+      return null;
+    }
+
+    String claimDrugCodeReformatted = null;
+
+    claimDrugCodeReformatted =
+        claimDrugCode.get().substring(0, 5) + "-" + claimDrugCode.get().substring(5, 9);
+
+    if (ndcProductHashMap.containsKey(claimDrugCodeReformatted)) {
+      String ndcSubstanceName = ndcProductHashMap.get(claimDrugCodeReformatted);
+      return ndcSubstanceName;
+    }
+
+    // log which NDC codes we couldn't find a match for in our downloaded NDC file
+    if (!drugCodeLookupMissingFailures.contains(claimDrugCode.get())) {
+      drugCodeLookupMissingFailures.add(claimDrugCode.get());
+      LOGGER.info(
+          "No national drug code value (PRODUCTNDC column) match found for drug code {} in"
+              + " resource {}.",
+          claimDrugCode.get(),
+          "fda_products_utf8.tsv");
+    }
```

Comment:
This leads to **so much** log noise, and no one has ever done anything with the data. I'd remove it or at least downgrade it to `debug`.